### DISCUSSION TITLE
Accessing properties on a PHP array of objects

### DISCUSSION THREAD

**Discussion by bszyman** [statamic/cms#5945] [Troubleshooting] [answered]
Hello Statafriends.

I'm adding a feed section to my website. Made a controller that curls out to a url and pulls a JSON feed in. When I parse and pass it into the (antlers) template, I can see the div containers for the ten entries being rendered out, but it seems like I can't access the properties inside of each of those ten objects.

I really don't see why this should work much differently than a collection of objects fetched from a collection within the CMS. Seems like an antlers oddity, but couldn't find much discussion about using standard php arrays within antlers to see if there are any gotchas to be aware of.

Controller:
```
$structured_results = json_decode($result)->posts;
$data = ["feed" => $structured_results];

return (new \Statamic\View\View)->template("home")->layout("layout")->with($data);
```

View:
```
{{ foreach:feed as="f" }}
    {{ f:user_info:name ensure_left="@" }}
    {{ f:created:date format="n/j/Y" }} at {{ f:created:date format="g:i a" }}
    {{ f:post_text }}
{{ /foreach:feed }}
```

Adding an {{ f | dump }} inside of the foreach dumps out all of the properties of the object, and all as expected.

![image](https://user-images.githubusercontent.com/2079918/165845877-bfda299d-75e7-4203-aea9-1198b4b0816c.png)

JSON comes in as such:

```
{
  "posts": [
    {
      "id": 101,
      "is_new": 0,
      "created": {
        ...
      },
      "updated": {
        ...
      },
      "post_text": "asdf ASDF asdf ASDF",
      "user": 6,
      "user_object": {
        "id": 0,
        "full_name": "",
      },
      "user_info": {
        "name": "---"
      },
      "published": 1,
      "public": 1
    },
    ...
  ]
}
```

**Comment by duncanmcclean** [comment, marked as answer]
What about if you try something like this:

```antlers
{{ feed }}
  {{ user_info:name }}
{{ /feed }}
```

Also: three back ticks at start and finish will give you proper code formatting (like my above one) 😉

**Reply by duncanmcclean** [reply]
Oh, you probably need to make that object an array instead, you can do that by providing a second parameter when decoding from JSON:

```php
json_decode($result, true)
```

Does that help?

**Reply by bszyman** [reply]
![image](https://user-images.githubusercontent.com/2079918/165847183-06bf26ee-970b-488d-bf2c-b07b90d93f1b.png)

What you wrote is how I usually would loop through a collection from the CMS, but it seems like there's something about PHP arrays (they are all associative, iirc) that is maybe making it an odd structure to work with inside of antlers.

**Reply by bszyman** [reply]
> json_decode($result, true)

Mmmmm that works!

For posterity:

Controller:
```
$structured_results = json_decode($result, true);
$data = ["feed" => $structured_results];

return (new \Statamic\View\View)->template("home")->layout("layout")->with($data);
```

View:
```
{{ feed:posts }}
    <div class="microblogPost">
        <div class="microblogPostHeader">
            <div class="microblogPostHeaderName">
                <span style="font-weight: bold;">{{ value:user_info:name ensure_left="@" }}</span>
            </div>
            <div class="microblogPostHeaderDate">
                <a href="https://howler.bszyman.com/post/{{ id }}/">{{ created:date format="n/j/Y" }} at {{ created:date format="g:i a" }}</a>
            </div>
        </div>

        <div class="microblogPostContent">
            {{ post_text }}
        </div>
    </div>
    {{ /feed:posts }}
```

**Reply by bszyman** [reply]
Appreciate you having a look and the sanity check!

**Reply by duncanmcclean** [reply]
No worries!
